### DISCUSSION TITLE
ERROR: java.lang.NoSuchMethodException: <unbound>=HashMap.add(Group); Errror with v 1.12.6

### DISCUSSION THREAD

**Discussion by jkjha** [oracle/opengrok#4347] [Q&A]
Trying to upgrade my opengrok version from 1.11.0 to 1.12.6, but getting below error and container seems to stuck at this stage for long...

Running version 1.12.6
synchronization period = 60 minutes
Deploying web application
API timeout set to 60 seconds
Checking if index matches current version
May 12, 2023 1:15:34 PM org.opengrok.indexer.configuration.Configuration read
INFO: Reading configuration from /opengrok/etc/configuration.xml
ERROR: java.lang.NoSuchMethodException: <unbound>=HashMap.add(Group);
Wiping out data root
Removing '/opengrok/data/xref'

I see there are some changes related to project grouping, so I tried deleting the read-only.xml file but still getting this error. 
I will attempt to downgrade to a previous version to determine if the issue persists at a specific version. Will keep this thread updated with more info. In the meantime, If you have any information, please let us know.

**Comment by vladak** [comment]
See release the notes for https://github.com/oracle/opengrok/releases/tag/1.12.1, it describes the changes necessary to group configuration.

**Reply by jkjha** [reply]
thanks for the link. I thought, if I modify read-only.xml, it will update the configuration.xml file. Because I was afraid of making changes to configuration.xml file as its a huge file. I will edit configuration.xml file and check.

**Reply by vladak** [reply]
If you change the read-only config only, you will have to run the indexer to generate the configuration with -W. Pre-existing complete configuration has to be changed by hand.
